tools/importer-rest-api-specs/components/parser/parse_data.go

```diff
@@ -81,19 +80,37 @@ func combineConstants(first map[string]resourcemanager.ConstantDetails, second m
 func combineModels(first map[string]models.ModelDetails, second map[string]models.ModelDetails) (*map[string]models.ModelDetails, error) {
 	output := make(map[string]models.ModelDetails)
 
-	for k, v := range first {
+	for k, v := range second {
 		output[k] = v
 	}
 
-	for k, v := range second {
+	for k, v := range first {
 		existing, ok := output[k]
 		if ok && len(existing.Fields) != len(v.Fields) {
 			return nil, fmt.Errorf("duplicate models named %q with different fields - first %d - second %d", k, len(existing.Fields), len(v.Fields))
```

Comment:
I will revert this since it was only needed when we were tried to link an implementation back to it's parent in this iteration. This has been pulled out now so it can go back to how it was before.